diff -Naur terraform.old/internal/command/init.go terraform.new/internal/command/init.go
--- terraform.old/internal/command/init.go
+++ terraform.new/internal/command/init.go
@@ -7,6 +7,7 @@
 	"context"
 	"errors"
 	"fmt"
+	"os"
 	"log"
 	"reflect"
 	"sort"
@@ -77,6 +78,11 @@
 	// -force-copy implies -migrate-state
 	if c.forceInitCopy {
 		c.migrateState = true
+	}
+
+	val, ok := os.LookupEnv("NIX_TERRAFORM_PLUGIN_DIR")
+	if ok {
+		initArgs.PluginPath = append(initArgs.PluginPath, val)
 	}
 
 	if len(initArgs.PluginPath) > 0 {
